docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+`Task Summary` in the `Menu` is clicked.
+
+The data of each chart will automatically rebind when user manage their tasks using the commands as below:
+
+* `add`
+* `done`
+* `delete`
+* `refresh`
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+Only when there is task status change after `refresh`, the charts will refresh accordingly.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+==== Example Usage Scenario
+Given below is an example usage scenario and how the task summary mechanism behaves at each step.
 
-image::UndoRedoState4.png[]
+Step 1. The user launches the application with some alive and archived task records in the `storage`.
+Statistics will perform on the records retrieved, and the result will be visualized as different charts organized in a `tab panel`.
+After the user clicks `Statistics -> Task Summary`, the `tab panel` will display.
+The user successfully executes `add`, `delete` or `done` command when any chart is on the screen.
+The diagram will automatically refresh and reflect the updated record.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+Step 2. The user successfully executes `add`, `delete` or `done` command when any chart is on the screen.
+The diagram will automatically refresh and reflect the updated record.
 
-image::UndoRedoState5.png[]
 
-The following activity diagram summarizes what happens when a user executes a new command:
+Step 3. The user executes `refresh` command, and there is a task status update after the execution.
+For example, when the user is playing with the application, and meanwhile, a task expires.
+The status won't update until user call `refresh` command. In this case, the chart will also automatically refresh and reflect the updated record.
+
+Step 4. The user executes `refresh`, and there is no task status update after the execution. The charts will remain the same.
+
+
+Activity diagram for step 4:
+
+image::TaskSummaryWithRefreshCommandActivityDiagram.png[width = "400", length = "600"]
```

Comment:
The conditions that come after "find any task with expired status" should be more explicit, maybe you can consider [has expired status], [else].

Also, remember to join the node back after your conditions are executed.